includes/compat/trait-wc-stripe-subscriptions.php

```
@@ -577,20 +579,26 @@ public function process_subscription_payment( $amount, $renewal_order, $retry =
 				}
 				try {
-					$subscriptions     = function_exists( 'wcs_get_subscriptions_for_renewal_order' )
+					$subscriptions = function_exists( 'wcs_get_subscriptions_for_renewal_order' )
 						? wcs_get_subscriptions_for_renewal_order( $renewal_order )
 						: [];
-					$terminal_statuses = [ 'cancelled', 'expired', 'trash', 'completed', OrderStatus::ON_HOLD ];
+
 					foreach ( $subscriptions as $subscription ) {
-						if ( in_array( $subscription->get_status(), $terminal_statuses, true ) ) {
-							continue;
+						if ( class_exists( 'WCS_Retry_Manager' ) && WCS_Retry_Manager::is_retry_enabled() ) {
```

PHPMD (2.15.0)

[error] Avoid using static access to class 'WCS_Retry_Manager' in method 'process_subscription_payment'. (undefined)

(StaticAccess)